include/matx/transforms/cub.h

```diff
@@ -1272,21 +1271,18 @@ void cub_min(OutputTensor &a_out, const InputOperator &a,
                             InputOperator,
                             detail::CUB_OP_REDUCE_MIN>::GetCubParams(a_out, a, stream);
 
-  // Get cache or new Sort plan if it doesn't exist
-  auto ret = detail::cub_cache.Lookup(params);
-  if (ret == std::nullopt) {
-    auto tmp = new detail::matxCubPlan_t<OutputTensor, InputOperator, detail::CUB_OP_REDUCE_MIN>{
-        a_out, a, {}, stream};
-
-    tmp->ExecMin(a_out, a, stream);
-    detail::cub_cache.Insert(params, static_cast<void *>(tmp));
-  }
-  else {
-    auto type =
-        static_cast<detail::matxCubPlan_t<OutputTensor, InputOperator, detail::CUB_OP_REDUCE_MIN> *>(
-            ret.value());
-    type->ExecMin(a_out, a, stream);
-  }
+  using cache_val_type = detail::matxCubPlan_t<OutputTensor, InputOperator, detail::CUB_OP_REDUCE_MIN, int>;
+  detail::GetCache().LookupAndExec<detail::cub_cache_t>(
+      detail::GetCacheIdFromType<detail::cub_cache_t>(),
+      params,
+      [&]() {
+        int test; ///\todo TYLER_TODO: this should not be needed
```

Comment:
Same

Reply:
I have another comment in this PR explaining why I needed this and asking for a second opinion on how to get around it. I'm not sure how to link to another comment in the same PR... it's at line [1276](https://github.com/NVIDIA/MatX/pull/694/files/1969207f7a78218fe53226b6d3cc0f211d68e2be#diff-7c667b17610a2198f407d07c43ae3ed0aa7f9d9777c0bd7d084df613b5d9d441)

Reply:
(I see now they were pending and I forgot to submit them).

Reply:
updated to remove int and use correct empty param based on our discussion